From c4ec6884c5df464a140a087276f69094309f4092 Mon Sep 17 00:00:00 2001
From: "Alexander F. Lent" <lx@xanderlent.com>
Date: Fri, 31 Jan 2025 22:45:01 -0500
Subject: [PATCH 2/3] Fix compiler flags.

It is a warning to use FORTIFY_SOURCE without optimization.
We are treating all warnings as errors, so always optimize.

Signed-off-by: Alexander F. Lent <lx@xanderlent.com>
---
 cmake/compiler_flags.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/compiler_flags.cmake b/cmake/compiler_flags.cmake
index 906f70b..c8829c9 100644
--- a/cmake/compiler_flags.cmake
+++ b/cmake/compiler_flags.cmake
@@ -29,6 +29,8 @@ if(NOT ${LINUX_SYSTEM_NAME} STREQUAL "cros_sdk")
     add_compile_options(
       # Enable macro security features
       -D_FORTIFY_SOURCE=2
+      # "FORTIFY_SOURCE requires compiling with optimization"
+      -O
     )
 endif()
 
-- 
2.48.1

